dev-libs/libstrophe: Added libstrophe, an XMPP library for C.

Chobbes opened the pull request:
libstrophe is a minimal XMPP library written in C. It has almost no
external dependencies, only an XML parsing library (expat or libxml
are both supported). It is designed for both POSIX and Windows
systems.

Closes: https://bugs.gentoo.org/255206
Package-Manager: Portage-2.3.62, Repoman-2.3.11

juippis commented on dev-libs/libstrophe/libstrophe-0.9.2.ebuild:
```diff
@@ -0,0 +1,21 @@
+
+EAPI=7
+
+inherit autotools
```
This eclass isn't being used in this ebuild.

Chobbes replied:
It worked without, but I thought it depended on autotools, and I thought this automatically added that as a dependency?

Or is that an implicit system dependency?

juippis replied:
> 
> Or is that an implicit system dependency?

Yes :) It's not a big thing and not really an error in any way, just keeping the ebuild cleaner.

Chobbes replied:
Great! It didn't look like an implicit system dependency to me, but I'm glad that it is! I'll remove that then :).

juippis commented on dev-libs/libstrophe/libstrophe-0.9.2.ebuild:
```diff
@@ -0,0 +1,21 @@
+	dev-libs/expat"
+
+RDEPEND="${DEPEND}"
+BDEPEND="${DEPEND}"
```
Nope, this is wrong. BDEPEND is a split from DEPEND, so their contents can't be the same :) BDEPEND are executable build-dependencies, like doxygen, gettext and pkg-config.

Chobbes replied:
Gotcha, so it should be `BDEPEND=""`?

juippis replied:
In this state, yes. However you will most likely want to add a doc use flag with doxygen dependency. Then it'd be `BDEPEND="doc? ( app-doc/doxygen )"`

juippis commented on dev-libs/libstrophe/metadata.xml:
```diff
@@ -0,0 +1,12 @@
+	<maintainer type="person">
+		<email>hobbes@ualberta.ca</email>
+		<name>Calvin beck</name>
+	</maintainer>
```
You'll need to add someone to commit for you. It's usually the proxy-maintainer project, but if you find a dev who commits for you, he needs to be added here.

Chobbes replied:
Hmmm. I must not have understood the proxy-maintainer thing I read. I'm not sure what you mean exactly. I'll give it another read.

juippis replied:
Check example from many other proxy-maintained ebuilds and you'll get the idea :)

juippis commented:
Then! 

I see optional configure settings, 
```
  --with-libxml2          use libxml2 for XML parsing, expat is the default
  --disable-tls           disable TLS support
```
that should be behind IUSE and managed in src_configure. I also see upstream makes it possible to generate and install HTML_DOCS during the build. Could you implement these in your ebuild?

Chobbes commented:
@juippis thanks for all of the comments. This is my first attempt at an ebuild, so it's all very much appreciated :).

I'll try to add those use-flags to the package.

I actually don't know anything about HTML_DOCS. Can you point me in the right direction for that?

As for rdeps, this is required for the profanity XMPP client (a popular console-based XMPP client). I have an ebuild for that which I hope to submit as well.

juippis commented:
> 
> I actually don't know anything about HTML_DOCS. Can you point me in the right direction for that?
> 

Well if you want to know theory behind, you'll need to read PMS and devmanual. How to implement it, something like this should work:
```
src_compile() {
	default

	if use doc ; then 
		doxygen . || die "Failed to generate docs with Doxygen."
	fi
}

src_install() {
	default
	use doc && local HTML_DOCS=( "${S}"/docs/html/. )
	einstalldocs
}
```
but please test it. `use doc && ( doxygen . || die "Message." )` should also work in place of that if-fi sentence, but I haven't tested it myself.

Chobbes commented:
@juippis I believe I have addressed all of your concerns. Please let me know if you think any other changes are necessary.

If not I'll get to work on getting my profanity ebuild up and running :).

Thanks for all of the help so far! Looking forward to making more ebuilds, so criticisms are much appreciated!

juippis commented on dev-libs/libstrophe/libstrophe-0.9.2.ebuild:
```diff
@@ -0,0 +1,40 @@
+LICENSE="GPL-3 MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="libxml2 +tls doc"
```
Switch `tls` to `ssl`. The description says "Add support for SSL/TLS connections (Secure Socket Layer / Transport Layer Security)"

I would switch libxml2 to expat, because usage of libxml2 is usually the default. Then you don't have to include any of the USE flags in metadata.xml file, because `doc expat ssl` are globally defined.

Also keep in alphabetical order, ie doc expat ssl.

juippis commented on dev-libs/libstrophe/libstrophe-0.9.2.ebuild:
```diff
@@ -0,0 +1,40 @@
+KEYWORDS="~amd64 ~x86"
+IUSE="libxml2 +tls doc"
+
+DEPEND="dev-libs/openssl:0=
```
Put behind `ssl? ( ... )`
(and in alphabetical order, under libxml or expat)

juippis commented on dev-libs/libstrophe/libstrophe-0.9.2.ebuild:
```diff
@@ -0,0 +1,40 @@
+
+DEPEND="dev-libs/openssl:0=
+	!libxml2? ( dev-libs/expat )
+	libxml2? ( dev-libs/libxml2 )"
```
This logic is good, but I would personally switch the USE flag from libxml2 to expat and switch this around. 

Whatever you decide to do, add a slot operator to libxml2, `dev-libs/libxml2:2`.

Chobbes commented:
@juippis thanks, made changes!

Chobbes commented:
@juippis perfect! I made that change as well.

Chobbes commented:
Hey @juippis, do I have to specifically poke proxy-maint, or will they automatically see the pull request?

juippis commented:
Yes, it seems like it's slipped through them (with 300 PRs open...)

One thing that could also help is push a program that uses this library, then link it to this PR (this needs to be merged before that), since people are quite reluctant in pushing new libraries without reverse deps into the tree.

juippis commented:
Seems like this has been added to the main tree. Sorry that the proxy-maint team couldnt do it in time. Please contact the new maintainer directly if you wish to help maintaining the package. We hope to see more of your contributions in the future, hopefully attended more hastily then :)
